src/modules/number/index.ts

```diff
@@ -150,7 +150,10 @@ export class NumberModule extends SimpleModuleBase {
         throw new FakerError(`Precision should be greater than 0.`);
       }
 
-      const factor = 1 / precision;
+      const logPrecision = Math.log10(precision);
+      const factor = Number.isInteger(logPrecision)
+        ? Math.pow(10, -logPrecision) //mathematically this is the same as 1/precision, however it helpfully gives integer values for all precisions of the form 10^-n
```

Comment:
Please move the comment to the front, maybe even another line up as it refers to the entire block. 

```suggestion
      // Workaround to get integer values for the reverse of all precisions of the form 10^-n
      const factor = Number.isInteger(logPrecision)
        ? (10 ** -logPrecision)
```

Also I prefer `**` over Math.pow (We should probably enable [prefer-exponentiation-operator](https://eslint.org/docs/latest/rules/prefer-exponentiation-operator) too)
Not sure whether the parenthesis are needed here.

Reply:
I created #2585 to enable that rule.

- #2585